js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillLogsTab.tsx

```diff
@@ -12,48 +12,62 @@ import {
 } from '../../ticks/InstigationEventLogTable';
 import {InstigationEventLogFragment} from '../../ticks/types/InstigationEventLogTable.types';
 
-const getResultsForBackfillLogsPage = (e?: BackfillLogsPageQuery) => {
-  return e?.partitionBackfillOrError.__typename === 'PartitionBackfill'
-    ? e.partitionBackfillOrError.logEvents.events
-    : [];
-};
-const getFetchStateForBackfillLogsPage = (e?: BackfillLogsPageQuery) => {
-  return e?.partitionBackfillOrError.__typename === 'PartitionBackfill'
-    ? e.partitionBackfillOrError.logEvents
-    : null;
+const getResultForBackfillLogsPage = (e: BackfillLogsPageQuery) => {
+  if (e.partitionBackfillOrError.__typename === 'PartitionBackfill') {
+    const {events, hasMore, cursor} = e.partitionBackfillOrError.logEvents;
+    return {data: events, hasMore, cursor, error: undefined};
+  } else {
+    return {data: [], hasMore: false, error: e.partitionBackfillOrError, cursor: undefined};
+  }
 };
 
 export const BackfillLogsTab = ({backfill}: {backfill: BackfillDetailsBackfillFragment}) => {
-  const {fetched: events, refreshState} = useCursorAccumulatedQuery<
+  const {
+    error,
+    fetched: events,
+    refreshState,
+  } = useCursorAccumulatedQuery<
     BackfillLogsPageQuery,
     BackfillLogsPageQueryVariables,
-    InstigationEventLogFragment
+    InstigationEventLogFragment,
+    {message: string}
   >({
     query: BACKFILL_LOGS_PAGE_QUERY,
-    variables: {backfillId: backfill.id},
-    getResultArray: getResultsForBackfillLogsPage,
-    getNextFetchState: getFetchStateForBackfillLogsPage,
+    variables: useMemo(() => ({backfillId: backfill.id}), [backfill]),
```

Comment:
I feel like this is a bit of a foot cannon to have to memoize because it's not required in useQuery -- think it's safe to JSON.stringify and then use the string value as a dep inside the hook, or is better to leave like this?  Couldn't decide 🤔

Reply:
I wonder how `useQuery` does it? JSON.stringify seems fine to me to improve dev ergonomics

Reply:
Sounds good!